client/layout/masterbar/logged-out.jsx

```diff
@@ -31,7 +31,7 @@ const MasterbarLoggedOut = ( { title, sectionName, translate } ) => (
 			}
 
 			{ 'login' !== sectionName
-			?	<Item url={ config( 'login_url' ) }>
+			?	<Item url={ config( 'login_url' ) + '?redirect_to=' + encodeURIComponent( window.location.href ) }>
```

Comment:
@klimeryk Since the desktop app doesn't provide a signed-out experience I think this should be okay, right?